tests/test_system_verilog_target.py

```diff
@@ -105,6 +107,26 @@ class X(m.Circuit):
         tester.compile_and_run(target, simulator=simulator)
 
 
+@pytest.mark.parametrize("use_packed_arrays", (False, True))
+def test_packed_arrays(use_packed_arrays):
+    name_ = f"test_system_verilog_target_packed_arrays_{use_packed_arrays}"
+
+    class _Foo(m.Circuit):
+        name = name_
+        T = m.Array[5, m.Array[12, m.Array[3, m.Bits[6]]]]
+        io = m.IO(I=m.In(T), O=m.Out(T))
+        io.O @= io.I
+
+    fault.Tester(_Foo).compile_and_run(
+        target="system-verilog",
+        simulator="vcs",
+        skip_compile=True,
+        skip_run=True,
+        use_packed_arrays=use_packed_arrays,
+    )
+    assert filecmp.cmp(f"gold/{name_}_tb.sv", f"build/{name_}_tb.sv")
```

Comment:
Can we add a test that double checks that at least pokes/peeks work with packed arrays from a functionality perspective?  There are some array tests in tests/test_tester/test_core.py that we could parametrize by the use_packed flag

Reply:
Ideally we'd test all the actions but I think poking/peeking would be a minimum

Reply:
@leonardt added peeks and pokes to the tests. Figure we want to check the generated testbench in the case that we can't run a SV simulator.

Reply:
also we can't use iverilog because it doesn't support nd arrays anyway